_source/_posts/2021-06-16-session-mgmt-react.md

```diff
@@ -0,0 +1,99 @@
+---
+layout: blog_post
+title: "A Developer's Guide to Session Management in React"
+author: nickolas-fisher
+by: contractor
+communities: [javascript]
+description: "A quick run-down of how to manage sessions in React."
+tags: [session, react]
+tweets:
+- "Need to learn how to manage sessions in React? We've got you covered!"
+- "Learn how to manage sessions in React!"
+- "Managing sessions in React can be chellenging. Check this out!"
+image: blog/featured/okta-react-skew.jpg
+type: awareness
+---
+
+Sessions can be a challenging topic for developers of all skill levels.  Many React developers never consider the internals of session management because so much of the work is abstracted away.  But, it is important to understand  what sessions are, how they work, and how best to manage and manipulate them.  
+
+There are several different strategies for session management in React.  In this article, you will learn the basics about sessions, how to manage them in React, and see some examples using common packages.  
+
+## Session Management Overview
+
+The first thing you should know is what exactly a "session" is.  In its simplest terms, a session is some data that is stored on the server.  The server then provides an ID to the client, which the client can use to make requests back to the server.  For example, if you needed access to a user's email address, you could store it against the session, then return an `ID` to the client.  The client could then request an email be sent using the `ID`, rather than passing his/her  email back to the server.  The  `ID` field is opaque, meaning that the client knows nothing about what is saved against the 'ID' field.  The `ID` can also contain validation and encryption.  And, the server can return client data, which would typically be encrypted, to the client that  he/she would frequently need.  
+In this article, I consider a range of server scenarios, as many times when working in React, developers don't have control over what the server does.  For example, when passing session data, the server will include the data in a cookie.  The server may also expect a cookie to be present when making a request.  Therefore, as a react developer, you will be required to include this cookie in your requests.  You may also choose to store the data in `localStorage`, however  then the server doesn't have access to it.  Local storage allows for more storage and can persist over browsing sessions, making it ideal for situations where you want to remember user actions across multiple browsing sessions.  
```

Comment:
Add a new line here so these paragraphs are separated. Otherwise, it's just a wall of text.